src/components/TrapFocus/TrapFocus.tsx

```diff
@@ -68,25 +78,25 @@ export class TrapFocus extends React.PureComponent<TrapFocusProps, State> {
   };
 
   private handleBlur = (event: FocusEvent) => {
-    const {relatedTarget} = event;
+    const {relatedTarget: currentTarget} = event;
     const {focusTrapWrapper} = this;
     const {trapping = true} = this.props;
 
-    if (relatedTarget == null || trapping === false) {
+    if (trapping === false) {
       return;
     }
 
     if (
       focusTrapWrapper &&
-      !focusTrapWrapper.contains(relatedTarget as HTMLElement) &&
-      !closest(relatedTarget as HTMLElement, '[data-polaris-overlay]')
+      !focusTrapWrapper.contains(currentTarget as HTMLElement)
     ) {
       event.preventDefault();
 
       if (event.srcElement === findFirstFocusableNode(focusTrapWrapper)) {
-        return focusLastFocusableNode(focusTrapWrapper);
+        return write(() => focusLastFocusableNode(focusTrapWrapper));
```

Comment:
For context, as I usually ask, why is RAF needed for this change?

Reply:
Without it, focus seems to leave the window content, despite the active element being the first/last focusable node.